### ISSUE TITLE
Change all Become* methods to be virtual

### ISSUE THREAD

**Issue by corneliutusnea** [akkadotnet/akka.net#2255]
It's a small change but I can then write my own CustomReceiveActor where I can override the become methods to do some custom logging (if I need it) when states transition.
This can improve debugging.

**Comment by Horusiath** [comment]
Why do you need virtual `Become` for that? Wouldn't that work as well?

```
class CustomReceiveActor : ReceiveActor {
    protected ILoggingAdapter Log { get; } = Context.GetLogger();
    protected void MyBecome(Action newBehavior) {
        Log.Debug("behavior changed");
        base.Become(newBehavior);
    }
}
```

**Comment by corneliutusnea** [comment]
Yes, I can use MyBecome, I could even use `protected new void Become(Action configure)` but both are just ways around an overprotective framework.
Make it easy for me as a developer to inherit the ReceiveActor and customize my behavior or handling of actions. Make everything easily extensible.

Without such changes I have to either write my own ReceiveActor (just to have my own code in the Become), make my own Become or reside on other tricks.

Similar for example the `protected sealed override void OnReceive` should not be sealed.
Why force me to use Receive and Become? What if I want to add generic custom logging or some verification in there? It feels overly restrictive with no real benefit.
